com.ibm.wala.core/src/com/ibm/wala/analysis/typeInference/TypeInference.java

```diff
@@ -727,8 +727,8 @@ public void visitStoreIndirect(SSAStoreIndirectInstruction instruction) {
 
     @Override
     public IVariable makeVariable(int valueNumber) {
-      SymbolTable st = ir.getSymbolTable();
       if (doPrimitives) {
+        SymbolTable st = ir.getSymbolTable();
```

Comment:
> Are these changes to `TypeInference` part of the fix for default methods?

No, they were just on my fork.

>  If not, can we pull all changes to TypeInference into a separate PR, to keep things clean?

I'm not quite sure how to "split" pull requests in GitHub. The best I can think of is to revert the performance improvement commits. Once you accept the PR, I'll unrevert them and issue another request.